clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -931,7 +929,24 @@ public void testMissingOffsetNoResetPolicy(GroupProtocol groupProtocol) {
 
         // lookup committed offset and find nothing
         client.prepareResponseFrom(offsetResponse(Collections.singletonMap(tp0, -1L), Errors.NONE), coordinator);
-        assertThrows(NoOffsetForPartitionException.class, () -> consumer.poll(Duration.ZERO));
+
+        if (groupProtocol == GroupProtocol.CONSUMER) {
+            // New consumer poll(ZERO) needs to wait for the offset fetch event add by a call to poll, to be processed
+            // by the background thread, so it can realize there are no committed offsets and then
+            // throw the NoOffsetForPartitionException
+            TestUtils.waitForCondition(() -> {
+                while (true) {
```

Comment:
We don't need this while-loop, since it is a part of `TestUtils.waitForCondition` already.

Reply:
Nice catch, thanks.

Reply:
yes, nice catch!